packages/react-ui/src/lib/data-selector-utils.ts

```diff
@@ -0,0 +1,153 @@
+export type MentionTreeNode = {
+  key: string;
+  data: {
+    propertyPath: string;
+    /**Key for json value */
+    displayName: string;
+    /**value for json key */
+    value?: string | unknown;
+    isSlice: boolean;
+    isTestStepNode?: boolean;
+  };
+  children?: MentionTreeNode[];
+};
+/**Traverses an object to find its child properties and their paths, stepOutput has to be an object on first invocation */
+function traverseStepOutputAndReturnMentionTree({
```

Comment:
I would split this into at least 3 functions for internal handing.

if (isObject) {
    // handle object
}
if (isArray) {
    // handle array function
}
return {

}